src/repr/src/adt/regex.rs

```diff
@@ -63,31 +62,35 @@ impl Regex {
     /// A simple constructor for the default setting of `dot_matches_new_line: true`.
     /// See <https://www.postgresql.org/docs/current/functions-matching.html#POSIX-MATCHING-RULES>
     /// "newline-sensitive matching"
-    pub fn new(pattern: String, case_insensitive: bool) -> Result<Regex, Error> {
+    pub fn new(pattern: &str, case_insensitive: bool) -> Result<Regex, Error> {
         Self::new_dot_matches_new_line(pattern, case_insensitive, true)
     }
 
     /// Allows explicitly setting `dot_matches_new_line`.
     pub fn new_dot_matches_new_line(
-        pattern: String,
+        pattern: &str,
         case_insensitive: bool,
         dot_matches_new_line: bool,
     ) -> Result<Regex, Error> {
-        let mut regex_builder = RegexBuilder::new(pattern.as_str());
+        let mut regex_builder = RegexBuilder::new(pattern);
         regex_builder.case_insensitive(case_insensitive);
         regex_builder.dot_matches_new_line(dot_matches_new_line);
         Ok(Regex {
-            pattern,
             case_insensitive,
             dot_matches_new_line,
             regex: regex_builder.build()?,
         })
     }
+
+    /// Returns the pattern string of the regex.
+    pub fn pattern(&self) -> &str {
+        self.regex.as_str()
```

Comment:
Could you put a comment here that `as_str` just returns the pattern string as given by the user, but (perhaps surprisingly) does not include any additional flags?